bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/BridgeHandler.java

```diff
@@ -397,6 +397,45 @@ private boolean getRooms() throws InterruptedException {
         }
     }
 
+    public Device getDeviceInfo(String deviceId)
+            throws BoschSHCException, InterruptedException, TimeoutException, ExecutionException {
+        @Nullable
+        BoschHttpClient httpClient = this.httpClient;
+        if (httpClient == null) {
+            throw new BoschSHCException("HTTP client not initialized");
+        }
+
+        String url = httpClient.getBoschSmartHomeUrl(String.format("devices/%s", deviceId));
+        Request request = httpClient.createRequest(url, GET);
+
+        ContentResponse contentResponse = request.send();
+
+        String content = contentResponse.getContentAsString();
+
+        int statusCode = contentResponse.getStatus();
```

Comment:
To you see a chance to write one common "sent" error handler to check the status and errorResponse?
We have similar code in ``getRooms`` and ``getDevices`` and for the discovery I again added similar checks.

Reply:
That's a good idea. I already added the BoschHttpClient.sendRequest method in the past, but then forgot about it. Plus it was a bit too unflexible. I extended the method a bit and use it now in BridgeHandler. You can check if it works in the places you mentioned, maybe we have to extend it even a bit more.